io/src/image_yuv422.cpp

```diff
@@ -48,12 +48,12 @@ using pcl::io::FrameWrapper;
 using pcl::io::IOException;
 
 pcl::io::ImageYUV422::ImageYUV422 (FrameWrapper::Ptr image_metadata)
-  : Image (image_metadata)
+  : Image (std::move(image_metadata))
 {}
 
 
 pcl::io::ImageYUV422::ImageYUV422 (FrameWrapper::Ptr image_metadata, Timestamp timestamp)
-  : Image (image_metadata, timestamp)
+  : Image (std::move(image_metadata), timestamp)
```

Comment:
I'm surprised `timestamp` was not eligible for the move operation as well. Do you know the reason? I see a couple more instances in the rest of the code in which in theory, it should also be applicable.

Reply:
`std::move()` makes only sense, in case there is anything, that can be moved. So in case the object is trivially copy constructible, clang does not add `std::move()`, because all bytes need to be copied. So easy to remember: In case there is any pointer within the object or it's children (e.g. `std::vector` contains pointer), `std::move` make sense. Otherwise not (if all members or members of children are primitive data types).

Reply:
Thank you for clarifying. I suspect it will be cumbersome to keep tabs, on what needs and doesn't need `std::move` when we are reviewing follow up PRs without the clang check on top. Trivial types are easy to see, but I suspect that forcontainer types it will become annoying to have to check every single definition to see if it has a pointer or not.

Reply:
Yep, we should integrate clang-tidy. But for this we need an newer image (clang-tidy-8, CMake 3.14) and time to write a parser for the result, as Azure doesn't have plugins (I believe).